### PR TITLE
[SYCL][ESIMD] Add ESIMD PrefixSum kernel tests.

### PR DESCRIPTION
**Pull request by kychendev**

### CODE REVIEW

File: SYCL/ESIMD/PrefixSum.cpp [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux && gpu && opencl
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out 20
```
Comment (vladimirlaz) [inline SYCL/ESIMD/PrefixSum.cpp]:
```suggestion
// RUN: HOST_RUN_PLACEHOLDER %t.out 20
```

File: SYCL/ESIMD/Prefix_Local_sum1.cpp [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux && gpu && opencl
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out 20
```
Comment (vladimirlaz) [inline SYCL/ESIMD/Prefix_Local_sum1.cpp]:
```suggestion
// RUN: HOST_RUN_PLACEHOLDER %t.out 20
```

File: SYCL/ESIMD/Prefix_Local_sum2.cpp [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux && gpu && opencl
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out 20
```
Comment (vladimirlaz) [inline SYCL/ESIMD/Prefix_Local_sum2.cpp]:
```suggestion
// RUN: HOST_RUN_PLACEHOLDER %t.out 20
```

File: SYCL/ESIMD/Prefix_Local_sum3.cpp [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux && gpu && opencl
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
```
Comment (vladimirlaz) [inline SYCL/ESIMD/Prefix_Local_sum3.cpp]:
```suggestion
// RUN: HOST_RUN_PLACEHOLDER %t.out
```

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: SYCL/ESIMD/PrefixSum.cpp [hunk 5]
```diff
@@ -0,0 +1,421 @@
...
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux && gpu && opencl
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: HOST_RUN_PLACEHOLDER %t.out 20
```
Comment (vladimirlaz) [inline SYCL/ESIMD/PrefixSum.cpp]:
```suggestion
// RUN: %HOST_RUN_PLACEHOLDER %t.out 20
```

File: SYCL/ESIMD/Prefix_Local_sum1.cpp [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux && gpu && opencl
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: HOST_RUN_PLACEHOLDER %t.out 20
```
Comment (vladimirlaz) [inline SYCL/ESIMD/Prefix_Local_sum1.cpp]:
```suggestion
// RUN: %HOST_RUN_PLACEHOLDER %t.out 20
```

File: SYCL/ESIMD/Prefix_Local_sum2.cpp [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux && gpu && opencl
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: HOST_RUN_PLACEHOLDER %t.out 20
```
Comment (vladimirlaz) [inline SYCL/ESIMD/Prefix_Local_sum2.cpp]:
```suggestion
// RUN: %HOST_RUN_PLACEHOLDER %t.out 20
```

File: SYCL/ESIMD/Prefix_Local_sum3.cpp [hunk 8]
```diff
@@ -0,0 +1,344 @@
...
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux && gpu && opencl
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: HOST_RUN_PLACEHOLDER %t.out
```
Comment (vladimirlaz) [inline SYCL/ESIMD/Prefix_Local_sum3.cpp]:
```suggestion
// RUN: %HOST_RUN_PLACEHOLDER %t.out
```

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: SYCL/ESIMD/PrefixSum.cpp [hunk 9]
```diff
@@ -0,0 +1,421 @@
...
+            cmk_prefix_iterative(buf, it.get_id(0), elem_stride, thrd_stride,
+                                 PREFIX_ENTRIES_LOW);
+          });
+    });
+  }
+}
+
+//************************************
+// Demonstrate summation of arrays both in scalar on CPU and parallel on device
+//************************************
```
Comment (kbobrovs) [inline SYCL/ESIMD/PrefixSum.cpp:376]: I suggest to add some differentiating comments into the beginning of each test, to give an idea what is the difference between them.

Comment (kychendev) [reply SYCL/ESIMD/PrefixSum.cpp:376]: ESIMD has only a few numbers of tests. As we developed PrefixSum, there were some intermediate steps that didn’t end up in the final implementation, but we like to turn them into ULT to increase our coverage of different usage cases.  They are smaller tests which allow compiler to easily isolate bugs.

We have added description to the ULT tests to summarize the difference in the new commit.

**Review by vladimirlaz — APPROVED** [review]

**Review by kbobrovs — APPROVED** [review]
LGTM.

**Pull request merged**